### PR TITLE
Update dotnet store examples and add a link for optimization step

### LINKED ISSUE #18316: dotnet store - explain usage of --framework option
According to this documentation there are three required parameters but not all of them are specified in the examples.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fc238c72-c1eb-a09f-5b7b-801722924d38
* Version Independent ID: dbfcef0a-1079-0a2c-6bda-d2b57445ec29
* Content: [dotnet store command - .NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-store)
* Content Source: [docs/core/tools/dotnet-store.md](https://github.com/dotnet/docs/blob/master/docs/core/tools/dotnet-store.md)
* Product: **dotnet-core**
* Technology: **dotnet-cli**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

### LINKED ISSUE #18317: dotnet store - explain --skip-optimization option
It is currently not clear what the "optimization phase" does so it is not clear what can be enabled/diabled by the --skip-optimization command line option

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fc238c72-c1eb-a09f-5b7b-801722924d38
* Version Independent ID: dbfcef0a-1079-0a2c-6bda-d2b57445ec29
* Content: [dotnet store command - .NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-store)
* Content Source: [docs/core/tools/dotnet-store.md](https://github.com/dotnet/docs/blob/master/docs/core/tools/dotnet-store.md)
* Product: **dotnet-core**
* Technology: **dotnet-cli**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

### LINKED ISSUE #27775: Wrong examples without required options for `dotnet store`
The Examples section contains wrong code
```
dotnet store --manifest packages.csproj --framework-version 2.0.0
```
and the second one
```
dotnet store --manifest packages.csproj --skip-optimization
```
returns the same error

>  error MSB4057: The target "ComposeStore" does not exist in the project.

It is because `dotnet store` has required options: `-f` and `-r`

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fc238c72-c1eb-a09f-5b7b-801722924d38
* Version Independent ID: dbfcef0a-1079-0a2c-6bda-d2b57445ec29
* Content: [dotnet store command - .NET CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-store)
* Content Source: [docs/core/tools/dotnet-store.md](https://github.com/dotnet/docs/blob/main/docs/core/tools/dotnet-store.md)
* Product: **dotnet-fundamentals**
* Technology: **dotnet-cli**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #27775
Fixes #18316
Fixes #18317

### CODE REVIEW

File: docs/core/tools/dotnet-store.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ dotnet store -h|--help
 
 - **`--skip-optimization`**
 
-  Skips the optimization phase.
+  Skips the optimization phase. For information about optimization, see [Preparing a runtime environment](../deploying/runtime-store.md#preparing-a-runtime-environment).
```
Comment (adegeo) [inline docs/core/tools/dotnet-store.md:60]: This should use the standard info text
```suggestion
  Skips the optimization phase. For more information about optimization, see [Preparing a runtime environment](../deploying/runtime-store.md#preparing-a-runtime-environment).
```

File: docs/core/tools/dotnet-store.md [hunk 2]
```diff
@@ -74,13 +74,13 @@ dotnet store -h|--help
 - Store the packages specified in the *packages.csproj* project file for .NET Core 2.0.0:
```
Comment (adegeo) [inline docs/core/tools/dotnet-store.md:74]: Intro text here says 2.0.0 but the command was updated for 6.0

**Review by adegeo — CHANGES_REQUESTED** [review]

File: docs/core/tools/dotnet-store.md (hunk 2 above)
Comment (tdykstra) [inline docs/core/tools/dotnet-store.md:74]:
```suggestion
- Store the packages specified in the *packages.csproj* project file for .NET 6.0.1:
```

**Review by adegeo — APPROVED** [review]

**Comment by tdykstra** [comment]
@adegeo thanks for the review!

**Pull request merged**